services/motion/builtin/move_request.go

```diff
@@ -14,15 +16,19 @@ import (
 	"go.viam.com/rdk/motionplan"
 	"go.viam.com/rdk/referenceframe"
 	"go.viam.com/rdk/resource"
+	"go.viam.com/rdk/robot/framesystem"
 	"go.viam.com/rdk/services/motion"
+	"go.viam.com/rdk/services/vision"
 	"go.viam.com/rdk/spatialmath"
 )
 
 // moveRequest is a structure that contains all the information necessary for to make a move call.
 type moveRequest struct {
-	config        *motion.MotionConfiguration
-	planRequest   *motionplan.PlanRequest
-	kinematicBase kinematicbase.KinematicBase
+	config             *motion.MotionConfiguration
+	planRequest        *motionplan.PlanRequest
+	kinematicBase      kinematicbase.KinematicBase
+	cameraVisionPairs  []map[resource.Name]vision.Service
```

Comment:
this shouldn't need to be a slice

Reply:
a singular camera can be used by more than one vision service.
Suppose we have camera named `A` and vision services named `B` and `C`.
Of this, valid obstacle detectors include `A & B` and `A & C`
Since a single key may lead us to more than one value I think this should remain as a slice of maps.

Reply:
Actually, this can become `map[resource.Name][]vision.Service` or `map[vision.Service][]resource.Name`.
I think the latter is better.